### ISSUE TITLE
AmazonClientException: Unable to instantiate request handler chain for client: null

### ISSUE THREAD

**Issue by tcastanedap** [aws/aws-sdk-java#1262]
Hello,

We are trying to connect to a S3 bucket using the AWS SDK for Java (aws-java-sdk-1.11.172.jar)
We are developing a webservice application (.war)  to deploy within a tomcat server.
The tomcat server is running at a EC2 instance (sao paulo)
We succesfully uploaded all the aws SDK dependencies to the tomcat folder and when we deploy our app they are succesfully loaded.

The AWS IAM credentials are also loaded within the server according to the documentation
http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-credentials.html

However, when our java app (loaded in tomcat) tries to create the S3 client (in each incoming request through our webservice), we get an exception.

This is how the code looks like
------------------
```
private AWSCredentials credentials;
private AmazonS3 s3;

credentials = new ProfileCredentialsProvider().getCredentials();
/* the credentials are ok if we print them after the line above*/

s3 = AmazonS3ClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(credentials)).build(); // <-- in this line we get the exception
```
-----------------
This is what the exception says:
 
`com.amazonaws.AmazonClientException: Unable to instantiate request handler chain for client: null`

And the stack trace:
-----------------
```
com.amazonaws.AmazonClientException: Unable to instantiate request handler chain for client: null
       at com.amazonaws.handlers.HandlerChainFactory.getGlobalHandlers(HandlerChainFactory.java:84)
        at com.amazonaws.services.s3.AmazonS3Client.init(AmazonS3Client.java:648)
        at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:631)
        at com.amazonaws.services.s3.AmazonS3Builder$1.apply(AmazonS3Builder.java:35)
        at com.amazonaws.services.s3.AmazonS3Builder$1.apply(AmazonS3Builder.java:32)
        at com.amazonaws.services.s3.AmazonS3ClientBuilder.build(AmazonS3ClientBuilder.java:64)
        at com.amazonaws.services.s3.AmazonS3ClientBuilder.build(AmazonS3ClientBuilder.java:28)
        at com.amazonaws.client.builder.AwsSyncClientBuilder.build(AwsSyncClientBuilder.java:46)
        ...
Caused by: java.lang.NullPointerException
        at com.amazonaws.handlers.HandlerChainFactory.getGlobalHandlers(HandlerChainFactory.java:64)
        ... 57 more
```
-----------------------

**Comment by spfink** [comment]
It looks like we aren't able to get a class loader in this case. If your application is loaded using the bootstrap class loader calls to getClassLoader will return null. We can add a null check in this case and just not load global handlers. I'll play around to see if we can find a way to load global handlers without using the class loader.

**Comment by shorea** [comment]
Looks like this may be an option.

https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getResource(java.lang.String)

**Comment by shorea** [comment]
A fix for this will be included in version 1.11.203.
